standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -9538,7 +9538,7 @@ private void writeMPartitionColumnStatistics(Table table, Partition partition,
     if (oldStats != null) {
       StatObjectConverter.setFieldsIntoOldStats(mStatsObj, oldStats);
     } else {
-      if (sqlGenerator.getDbProduct().isPOSTGRES() && mStatsObj.getBitVector() == null) {
```

Comment:
you could aslo move this into the `setBitVector` / defaults stuff into the `MTableColumnStatistics`

Reply:
done